stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/MessageDispatcher.java

```diff
@@ -221,17 +232,69 @@ private void handleError(byte[] message, long deliveryTag, AMQP.BasicProperties
+            RabbitMQMetricsUtil.reportError(channel, RabbitMQObservabilityConstants.ERROR_TYPE_CONSUME);
             throw new RabbitMQConnectorException(RabbitMQConstants.THREAD_INTERRUPTED);
         } catch (AlreadyClosedException | BallerinaConnectorException exception) {
+            RabbitMQMetricsUtil.reportError(channel, RabbitMQObservabilityConstants.ERROR_TYPE_CONSUME);
             throw new RabbitMQConnectorException("Error occurred in RabbitMQ service. ");
         }
     }
 
+    private void executeResource(CountDownLatch countDownLatch, byte[] message, long deliveryTag,
+                                 AMQP.BasicProperties properties) {
+        if (ObserveUtils.isTracingEnabled()) {
+            Map<String, Object> observerProperties = getNewObserverContextInProperties();
+            runtime.invokeMethodAsync(service, RabbitMQConstants.FUNC_ON_MESSAGE,
+                                      new RabbitMQResourceCallback(countDownLatch, channel, queueName, message.length),
+                                      observerProperties, getMessageObjectValue(message, deliveryTag, properties),
+                                      true);
+            return;
+        }
+        runtime.invokeMethodAsync(service, RabbitMQConstants.FUNC_ON_MESSAGE,
+                                  new RabbitMQResourceCallback(countDownLatch, channel, queueName, message.length),
+                                  getMessageObjectValue(message, deliveryTag, properties), true);
+    }
+
+    private void executeResource(CountDownLatch countDownLatch, byte[] message, ObjectValue messageObjectValue,
+                                 Object forContent) {
+        if (ObserveUtils.isTracingEnabled()) {
+            Map<String, Object> observerProperties = getNewObserverContextInProperties();
+            runtime.invokeMethodAsync(service, RabbitMQConstants.FUNC_ON_MESSAGE,
+                                      new RabbitMQResourceCallback(countDownLatch, channel, queueName, message.length),
+                                      observerProperties, messageObjectValue, true, forContent, true);
+            return;
+        }
+        runtime.invokeMethodAsync(service, RabbitMQConstants.FUNC_ON_MESSAGE,
+                                  new RabbitMQResourceCallback(countDownLatch, channel, queueName, message.length),
+                                  messageObjectValue, true, forContent, true);
+    }
+
+    private void executeOnErrorResource(CountDownLatch countDownLatch, ObjectValue messageObjectValue, ErrorValue error,
+                                        AMQP.BasicProperties properties) {
+        if (ObserveUtils.isTracingEnabled()) {
+            Map<String, Object> observerProperties = getNewObserverContextInProperties();
+            runtime.invokeMethodAsync(service, RabbitMQConstants.FUNC_ON_ERROR,
+                                      new RabbitMQErrorResourceCallback(countDownLatch), observerProperties,
```

Comment:
```suggestion
                                      new RabbitMQErrorResourceCallback(countDownLatch), getNewObserverContextInProperties(),
```
As the variable has no any other usages.